narwhals/_pandas_like/dataframe.py

```diff
@@ -233,12 +236,28 @@ def select(
         )
         return self._from_native_frame(df)
 
-    def drop_nulls(self, subset: str | list[str] | None) -> Self:
+    def drop_nulls(
+        self: Self, subset: str | Selector | list[Selector | str] | None
+    ) -> Self:
         if subset is None:
             return self._from_native_frame(self._native_frame.dropna(axis=0))
-        subset = [subset] if isinstance(subset, str) else subset
+
+        from narwhals.selectors import Selector
+
         plx = self.__narwhals_namespace__()
-        return self.filter(~plx.any_horizontal(plx.col(*subset).is_null()))
+        subset_: Iterable[str | Selector] = (
+            [subset] if isinstance(subset, (str, Selector)) else subset
+        )
+
+        column_names = [c for c in subset_ if isinstance(c, str)]
+        selectors = [c.is_null()._call(plx) for c in subset_ if isinstance(c, Selector)]
```

Comment:
do we need to call `is_null` here if we're not using these `selectors` in `any_horizontal`?

Reply:
I am not sure I fully get the question, this part:

> if we're not using these selectors in any_horizontal?

how so?

My reasoning is as follows:

- If a selector is present, we evaluate it, else this will end up being an empty list anyway.
- `any_horizontal` is called only if at least one (evaluated) selector ends up being there

Reply:
oops sorry, I meant to comment this on the `_dask` one - if you only need to get the column names, out, does it work to just do `*[c._call(plx) for c in subset_ if isinstance(c, Selector)]` there?

Reply:
Oh that seems significantly better! Let me give it a try